crates/polars-time/src/chunkedarray/datetime.rs

```diff
@@ -129,7 +129,20 @@ pub trait DatetimeMethods: AsDatetime {
             TimeUnit::Microseconds => datetime_to_ordinal_us,
             TimeUnit::Milliseconds => datetime_to_ordinal_ms,
         };
-        ca.apply_kernel_cast::<Int16Type>(&f)
+        match ca.dtype() {
+            #[cfg(feature = "timezones")]
+            DataType::Datetime(_, Some(_)) => {
+                let ca_local = polars_ops::chunked_array::replace_time_zone(
+                    ca,
+                    None,
+                    &StringChunked::new("".into(), ["raise"]),
+                    NonExistent::Raise,
+                )
+                .expect("Removing time zone is infallible");
+                ca_local.apply_kernel_cast::<Int16Type>(&f)
+            },
+            _ => ca.apply_kernel_cast::<Int16Type>(&f),
```

Comment:
Can we return `ca` from the match statement?

Then we only have to apply `&f` once, saving some compiler bloat.

Reply:
Will already merge so it's included in release. Could you make a follow up?